src/theme.tsx

```diff
@@ -1,33 +1,86 @@
-import { createTheme } from "@mui/material";
+import { Theme, ToggleButtonGroup, createTheme, styled } from "@mui/material";
 import NunitoSans from "../resources/fonts/NunitoSans.ttf";
 
-export const theme = createTheme({
+export const theme: Theme = createTheme({
+  palette: {
+    primary: {
+      main: "#222"
+    },
+    secondary: {
+      main: "#f78da7"
+    }
+  },
   typography: {
-    fontFamily: 'Nunito Sans',
-    fontWeightRegular: 300,
+    fontFamily: "Nunito Sans",
+    fontWeightRegular: 400,
     fontSize: 15
   },
   components: {
     MuiCssBaseline: {
       styleOverrides: `
         @font-face {
           font-family: 'Nunito Sans';
-          font-style: sans-serif;
-          font-display: swap;
-          font-weight: 300;
-          font-size: 2em;
-          src: local('NunitoSans'), url(${NunitoSans}) format('ttf');
-          unicodeRange: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF;
+          src: local('NunitoSans'), url(${NunitoSans}) format('truetype');
         }
-      `,
+      `
     },
-  },
-  palette: {
-    primary: {
-      main: '#222',
+    MuiAppBar: {
+      defaultProps: {
+        color: "default"
+      },
+      styleOverrides: {
+        root: {
+          top: "1em",
+          borderRadius: 15
+        }
+      }
     },
-    secondary: {
-      main: "#f78da7"
+    MuiTooltip: {
+      defaultProps: {
+        arrow: true
+      }
+    },
+    MuiAvatar: {
+      styleOverrides: {
+        root: {
+          height: 48,
+          width: 48
+        }
+      }
+    },
+    MuiCard: {
+      defaultProps: {
+        elevation: 4
+      },
+      styleOverrides: {
+        root: {
+          backgroundColor: "#f5f5f5"
+        }
+      }
     }
+  },
+  shape: {
+    borderRadius: 15
   }
 });
+
+/**
+ * Styled toggle button group component for language switching
+ */
+export const LanguageButtons = styled(ToggleButtonGroup)(({ theme }) => ({
```

Comment:
I haven't seen styled components placed within the theme before, usually they go in the component file.

Reply:
reverted